server/helpers/index.js

```
@@ -1,6 +1,189 @@
+    https.get(
+      `https://www.codewars.com/api/v1/users/${userName}/code-challenges/completed`,
+      (httpsResponse) => {
+        try {
+          if (httpsResponse.statusCode !== 200) {
+            throw new Error(`
+              Request Failed.
+              Status Code: ${httpsResponse.statusCode}
+              API: code-challenge.
+              User: ${userName}
+              Page: 0
+            `)
+          }
+        } catch (err) {
+          // Log err
+          reject()
+        }
+        let data = ''
+
+        httpsResponse.on('data', (chunk) => {
+          data += chunk
+        })
+
+        httpsResponse.on('end', () => {
+          try {
+            const response = JSON.parse(data)
+            if (response.success === false) {
+              throw new Error('User not found')
+            }
+
+            // Adds completed kata ids to an array
+            let completed = response.data.map(kata => kata.id)
+            if (response.totalPages === 1) {
+              resolve(completed)
+            } else {
+              // If there is more than one page of results all other pages must
+              // also be requested
+              getHistoricCompletedKatas(userName, response.totalPages)
+                .then((historicCompleted) => {
+                  completed = [...completed, ...historicCompleted]
+                  resolve(completed)
+                })
+                .catch((err) => {
+                  reject(err)
+                })
+            }
+          } catch (err) {
+            // If the user is not found it may indicate that their account has
+            // been deleted
+            // Log err
+            reject()
+          }
+        })
+      }
+    )
+  })
+}
+
+export const updateAllScores = async (katathonId) => {
+  try {
+    const katathon = await Katathon.findById(katathonId)
+    const { users, katas } = katathon
+    if (!users.length || !katas.length) {
+      return
```

Do we want to throw an error here, rather than returning undefined?